services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/DataPlaneClientImpl.java

```diff
@@ -48,17 +49,21 @@
     @Autowired
     LocalCache localCache;
 
+    @Autowired
+    NodeTopicCache nodeTopicCache;
+
     private Map<String, List<String>> getMulticastTopics(List<String> hostIps) throws Exception {
         Map<String, List<String>> multicastTopics = new HashMap<>();
 
         for (String hostIp : hostIps) {
-            String groupTopic = localCache.getNodeInfoByNodeIp(hostIp).get(0).getGroupTopic();
+            String hostId = localCache.getNodeInfoByNodeIp(hostIp).get(0).getId();
+            String groupTopic = nodeTopicCache.getNodeTopicInfo(hostId).getGroupTopic();
```

Comment:
This is across two caches/tables. We will make it in a transaction.

Reply:
Apply change to other places that involve multiple cache access.

Reply:
+1

Reply:
Alright.
There might be a few more places like this. I haven't touched these parts which is why I hadn't noticed. It will be a great help if and when anyone of us notice code requring transaction, to notify all of us.
I will try to sweep through the code for more points like this.

Reply:
I have tried to use @Transactional annotation to make it in a transaction. And same changes have made for the below comments.